src/consensus/replay_tower.zig

```diff
@@ -1579,6 +1613,622 @@ fn optimisticallyBypassVoteStakeThresholdCheck(
     return false;
 }
 
+pub fn collectVoteLockouts(
+    allocator: std.mem.Allocator,
+    logger: Logger,
+    vote_account_pubkey: *const Pubkey,
+    bank_slot: Slot,
+    vote_accounts: *const StakeAndVoteAccountsMap,
+    ancestors: *const AutoHashMapUnmanaged(Slot, SortedSet(Slot)),
+    progress_map: *const ProgressMap,
+    latest_validator_votes_for_frozen_banks: *LatestValidatorVotesForFrozenBanks,
+) !ComputedBankState {
+    var vote_slots = SortedSet(Slot).init(allocator);
+    defer vote_slots.deinit();
+
+    var voted_stakes = VotedStakes.empty;
+
+    var total_stake: u64 = 0;
+
+    // Tree of intervals of lockouts of the form [slot, slot + slot.lockout],
+    // keyed by end of the range
+    var lockout_intervals = LockoutIntervals.empty;
+    errdefer lockout_intervals.deinit(allocator);
+
+    var my_latest_landed_vote: ?Slot = null;
+
+    for (vote_accounts.keys(), vote_accounts.values()) |key, value| {
+        const voted_stake = value.stake;
+        const vote_account = value.account;
+        // Skip accounts with no stake.
+        if (voted_stake == 0) {
+            continue;
+        }
+
+        logger.trace().logf(
+            "{} {} with stake {}",
+            .{ vote_account_pubkey, key, voted_stake },
+        );
+
+        var vote_state = try TowerVoteState.fromAccount(
+            &vote_account,
+        );
+
+        for (vote_state.votes.constSlice()) |vote| {
+            const interval = try lockout_intervals
+                .getOrPut(allocator, vote.lastLockedOutSlot());
+            if (!interval.found_existing) {
+                interval.value_ptr.* = .init(allocator);
+            }
+            try interval.value_ptr.append(.{ .slot = vote.slot, .pubkey = key });
+        }
+
+        // Vote account for this validator
+        if (key.equals(vote_account_pubkey)) {
+            my_latest_landed_vote = if (vote_state.nthRecentLockout(0)) |l| l.slot else null;
+            logger.debug().logf("vote state {any}", .{vote_state});
+            const observed_slot = if (vote_state.nthRecentLockout(0)) |l| l.slot else 0;
+
+            logger.debug().logf("observed slot {any}", .{observed_slot});
```

Comment:
why are we logging this? debug is the default log level fyi

Reply:
Actually this made me realise the the removal from the `tower.zig` must have sneaked back in, during one of the merges from the target branch. I'll update that now

Reply:
Removed the moved definitions here [ea876b2](https://github.com/Syndica/sig/pull/798/commits/ea876b2e19eca15689fba04f409dcc480689b155)